### ISSUE TITLE
Magento 2.3 - Issue to override price-box.js using mixin(In Firefox 66.0.2 version)

### ISSUE THREAD

**Issue by miteshdhaduk** [magento/magento2#22338]
### Preconditions (*)

1. Magento 2.3 (Firefox 66.0.2 (64-bit) version)
2. Magento 2.3.1 (Firefox 66.0.2 (64-bit) version)
3. Magento 2.2.8 (Firefox 66.0.2 (64-bit) version)

### Steps to reproduce (*)

1. Override updatePrice function(Magento_Catalog/js/price-box) using a mixin.
2. File call using mixin properly but the updatePrice function is not called. (In chrome working fine, but in Firefox not working.)

### Expected result (*)

1. updatePrice function is call properly in all browser.

### Actual result (*)

1. updatePrice function is not call in Firefox.

**Comment by raumatbel** [comment]
We are making a mix of priceBox and exactly the same thing happens to us. In Chrome it works but in Firefox it does not.

**Comment by miteshdhaduk** [comment]
@raumatbel Have you fix it or not?

**Comment by lionelalvarez** [comment]
Hello, I have had the same problem in Magento 2.2.6

**Comment by miteshdhaduk** [comment]
> Hello, I have had the same problem in Magento 2.2.6

Have you fix it or not?

**Comment by raumatbel** [comment]
At this time, the problem is not fixed @miteshdhaduk

**Comment by mageprashant** [comment]
@lionelalvarez @miteshdhaduk 
It's my code and working fine to me.

> requirejs-config.js
```
var config = 
{
    config: 
    {
        mixins: 
        {
            'Magento_Catalog/js/price-box':
            {
            	'Vendor_Module/js/price-box': true
            }
        }
    }
};
```

> price-box.js
```
define([
	'jquery',
	'Magento_Catalog/js/price-utils',
	'underscore',
	'mage/template',
	'jquery/ui'
], function ($, utils, _, mageTemplate) 
{
	'use strict';
	
	return function (widget) 
	{
		$.widget('mage.priceBox', widget, 
		{
			updatePrice: function updatePrice(newPrices) {
			{
				// Custom code
			}
		});
		
		return $.mage.priceBox;
	}
});
```

**Comment by miteshdhaduk** [comment]
@mageprashant Yes, you are right. But this is not working in Magento 2.2.8 and Magento 2.3.*(Not working in Firefox).

**Comment by niravkrish** [comment]
> @lionelalvarez @miteshdhaduk
> It's my code and working fine to me.
> 
> > requirejs-config.js
> 
> ```
> var config = 
> {
>     config: 
>     {
>         mixins: 
>         {
>             'Magento_Catalog/js/price-box':
>             {
>             	'Vendor_Module/js/price-box': true
>             }
>         }
>     }
> };
> ```
> > price-box.js
> 
> ```
> define([
> 	'jquery',
> 	'Magento_Catalog/js/price-utils',
> 	'underscore',
> 	'mage/template',
> 	'jquery/ui'
> ], function ($, utils, _, mageTemplate) 
> {
> 	'use strict';
> 	
> 	return function (widget) 
> 	{
> 		$.widget('mage.priceBox', widget, 
> 		{
> 			updatePrice: function updatePrice(newPrices) {
> 			{
> 				// Custom code
> 			}
> 		});
> 		
> 		return $.mage.priceBox;
> 	}
> });
> ```

Code Is working Fine with Magento 2.3.
I am also check in Firefox Browser.

**Comment by miteshdhaduk** [comment]
@mageprashant @niravkrish Yes you are right. But this is not work in Firefox 66.0.2 (64-bit) version.

**Comment by miteshdhaduk** [comment]
@ajijshekh123 Why you closed it?

**Comment by ajijshekh123** [comment]
Sorry, @miteshdhaduk It was by mistake close the issue.

**Comment by ghost** [comment]
Hi @miteshdhaduk thank you for you report, iєm not able to reproduce followings steps you described on clean magento 2.3-develop and Firefox 66.0.2 (64-bit) version. please refer to -> [Community Forums](https://community.magento.com/) or the [Magento Stack Exchange](https://magento.stackexchange.com/)

**Comment by miteshdhaduk** [comment]
@engcom-backlog-nazar I used php7.2.17. And currently I also checked and it is not working. I override updatePrice function using mixin. File execute but the function is not called.

**Comment by DeepPopat** [comment]
Hello @engcom-backlog-nazar 

I have checked this issue with the fresh installation (Magento 2.3.*) in my local system. with the Firefox latest version. I have also attached the gif over here.

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/32391312/56454404-a0b03b80-636d-11e9-9842-4f6f900d3f47.gif)

**Comment by ghost** [comment]
Hi @DeepPopat so please provide exact steps to reproduce not a gif.

**Comment by miketaylr** [comment]
Hi all - Firefox engineer here. In 66 we added support for `window.event` and changed the value of `keyCode` inside a `keypress` event handler (to match what Chrome does). So rather than be 0, it will be set to the same value as charCode.

We also don't dispatch keypress for non-printable keys: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent#Compatibility_notes

So if you had Firefox-specific keypress handling, you should be able to just use whatever you were sending to Chrome for 66+. For older versions, you can use the absence of window.event as a signal it will need the legacy behavior.

I'm not familiar with magento code, or if this component uses keypress... but I'm happy to help if someone who is more familiar with magento can help me. :)

**Comment by AlexWorking** [comment]
Hi, @miteshdhaduk. We are closing this issue due to inactivity.

**Comment by rich-humphreys** [comment]
I'd just encountered this issue myself, attempting to put a mixin on `price-box.js`. The mixin was working fine in Chrome, however not in Firefox. I'd advise changing the single quotes within the `requirejs-config.js` file to double quotes, clearing your cache and testing again.

**Comment by pravalitera** [comment]
@AlexWorking 
I understand that you need to clean the backlog, but really. Stop closing every issue wih "no activity" or "i can't reproduce it". It's fucking not true. I've already read the same exact pb on 2 anothers tickets :

https://github.com/magento/magento2/issues/7322
https://github.com/magento/magento2/issues/14118 

It's really annoying when you encounter a real problem, to see that it's as nobody cares about community...


edit: and hooray, a guy found a "fix"
https://magento.stackexchange.com/questions/267502/price-box-js-overriding-via-mixin-bug-magento-2-3
edit2: i had to put another selector than the one written in the stackExchange, but definetly, the mixin is working properly this way.
But still, does not work the way it should work...

**Comment by srenon** [comment]
@miteshdhaduk... is this issue fixed?

This should fix the filtering issue

    <script type="text/x-magento-init">
        {
            "[data-role=priceBox][data-price-box=product-id-<?= $block->escapeHtml($_product->getId()) ?>]": {
                "Magento_Catalog/js/price-box": {
                    "priceConfig":  <?= /* @escapeNotVerified */ $block->getJsonConfig() ?>
                }
            }
        }
    </script>

See /app/code/Magento/Swatches/view/frontend/templates/product/listing/renderer.phtml and https://github.com/magento/magento2/pull/12929

**Comment by magento-engcom-team** [comment]
Hi @miteshdhaduk. Thank you for your report. 
The issue has been fixed in magento/magento2#24054 by @srenon in 2.3-develop branch
Related commit(s):
 - [b30deddc054d936b2c3eab1832cc07f74de62a41](https://github.com/magento/magento2/commit/b30deddc054d936b2c3eab1832cc07f74de62a41)
 - [f9b1f0b8336b4f8d870aaed55a83645ab0a7386c](https://github.com/magento/magento2/commit/f9b1f0b8336b4f8d870aaed55a83645ab0a7386c)
 - [fb350e66ca81105698c2bbb2fcae4821b061ab8d](https://github.com/magento/magento2/commit/fb350e66ca81105698c2bbb2fcae4821b061ab8d)

The fix will be available with the upcoming 2.3.4 release.

**Comment by tzyganu** [comment]
Not sure if this counts as a valid solution, but it worked for me. I was trying to add soemthing to the `reloadPrice` function and I got the same error as everyone else.  
This :point_down: resulted in the error. 
```
define(
    ['jquery'],
    function ($) {
        return function (originalPriceBox) {
            $.widget('myNs.priceBox', originalPriceBox, {
                reloadPrice: function reDrawPrices()
                {
                    this._super();
                    //execute something after `reloadPrice`
                }
            });
            return $.myNs.priceBox;
        };
    }
);
```

This :point_down: worked nicely

```
